### PR TITLE
589 validate objects coming via api

### LINKED ISSUE #589: Validate objects coming via API in BE via the integrated validator
1) remove todo  on line 33 in file namespace.py - we will have only validation via route_utils request_body function
2) check that everywhere we are using request_body function to get object from request body (we dont do it for example in matching_api.py
3) on line 234 in error handler.py there is logger.error(f'Unexpected error code returned {error_code}, returning 500 instead') but it does not make sense there, put this logging to correct place
4) add validations to all objects we are using request_body function for. In some cases we do it (for example for patients we do not allow height negative etc. But for example for configuration we do not do that. Add the validation there as well. (use fronted or ask if you are not sure what the valid values are)

### PR DESCRIPTION
**Pull request by krllstdn**
Closes #589

### CODE REVIEW

**Review by kristinagalik — CHANGES_REQUESTED** [review]

File: txmatching/configuration/configuration.py [hunk 1]
```diff
@@ -10,3 +12,13 @@ class Configuration:
...
+
+    def __post_init__(self):
+        if not does_configuration_exist(self.id):
+            raise ValueError(f"Configuration with id {self.id} does not exist")
+
+        does_txm_event_exist(self.txm_event_id)
+
+
+def does_configuration_exist(configuration_id):
+    return ConfigModel.query.filter_by(id=configuration_id).first() is not None
```
[the remark this answers is not in the document]
Comment (krllstdn) [reply txmatching/configuration/configuration.py:24]: @kristinagalik It has to be, but if it is placed there the _circular import_ occurs

File: txmatching/configuration/configuration.py [hunk 2]
```diff
@@ -10,3 +13,13 @@ class Configuration:
...
     txm_event_id: int
     parameters: ConfigParameters
+
+    def __post_init__(self):
+        raise_error_if_configuration_does_not_exist(self.id)
+
+        raise_error_if_txm_event_does_not_exist(self.txm_event_id)
+
+# it's not possible to place this function in config_service.py because of circular import
+def raise_error_if_configuration_does_not_exist(configuration_id):
```
Comment (kristinagalik) [inline txmatching/configuration/configuration.py:23]: duplicate definition of this function

File: txmatching/configuration/config_parameters.py [hunk 3]
```diff
@@ -124,3 +125,22 @@ def comparable(self, other):
                         return False
 
         return True
+
+    # pylint: disable=too-many-branches
+    def __post_init__(self):
```
Comment (kubantjan) [inline txmatching/configuration/config_parameters.py:130]: I would add non negativnes as a property of the field (such as COMPARISON_MODE property, so we do not have to do this repetitive code here and simply iterate through the fields)

File: txmatching/database/services/patient_service.py [hunk 4]
```diff
@@ -412,16 +412,21 @@ def delete_donor_recipient_pair(donor_id: int, txm_event_id: int):
...
                                        f'than minimal required height {recipient_requirements.min_donor_height}.')
 
     if (recipient_requirements.min_donor_weight and recipient_requirements.max_donor_weight and
-        recipient_requirements.min_donor_weight > recipient_requirements.max_donor_weight):
+            recipient_requirements.min_donor_weight > recipient_requirements.max_donor_weight):
         raise InvalidArgumentException(f'Maximal required weight {recipient_requirements.max_donor_weight} is smaller '
                                        f'than minimal required weight {recipient_requirements.min_donor_weight}.')
+
+
+def raise_error_if_donor_in_txm_event_doesnt_exist(txm_event_id: int, donor_id: int):
```
Comment (kubantjan) [inline txmatching/database/services/patient_service.py:430]: I do not see here how you check that the donor is in the txm_event

**Review by kubantjan — CHANGES_REQUESTED** [review]
overall ok, small comments

File: txmatching/database/services/patient_service.py (hunk 4 above)
Comment (krllstdn) [reply txmatching/database/services/patient_service.py:430]: Hmm, interesting, the check disappeared. WIll be corrected

**Review by kristinagalik — DISMISSED** [review]

File: tests/database/services/test_configuration.py [hunk 5]
```diff
@@ -107,6 +107,22 @@ def test_configuration_comparison(self):
         self.assertFalse(
             ConfigParameters(required_patient_db_ids=[1]).comparable(ConfigParameters(required_patient_db_ids=[1, 2])))
 
+    def test_configuration_non_negative_parameter(self):
+        with self.assertRaises(ValueError):
+            ConfigParameters(minimum_total_score=-1)
```
[the remark this answers is not in the document]
Comment (krllstdn) [reply tests/database/services/test_configuration.py:112]: @kristinagalik So you mean that it makes sense to leave only one check of the first parameter?

[the remark this answers is not in the document]
Comment (kubantjan) [reply tests/database/services/test_configuration.py:112]: @kristinagalik  is right. 

I would say that better is to check for all (as you have it here ready already) but do the chceck for valueerrro separately)

**Review by kubantjan — CHANGES_REQUESTED** [review]
tiny comment otherwise alright

File: tests/database/services/test_configuration.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (krllstdn) [reply tests/database/services/test_configuration.py:112]: thanks for the quick reply

**Review by kubantjan — APPROVED** [review]

**Pull request merged**